RealmSwift/RealmProperty.swift

```diff
@@ -81,7 +79,17 @@ extension RealmProperty: Codable where Value: Codable {
 }
 
 /// A protocol describing types that can parameterize a `RealmPropertyType`.
-public protocol RealmPropertyType {}
+public protocol RealmPropertyType {
+    static func nilValue() -> Self
```

Comment:
Can this be combined with the existing `_nilValue()`?

Reply:
I have moved `_nilValue` to `_RealmSchemaDiscoverable`